ml-agents/mlagents/trainers/trajectory.py

```diff
@@ -11,7 +11,25 @@
 from mlagents.trainers.torch.action_log_probs import LogProbsTuple
 
 
+class AgentStatus(NamedTuple):
+    """
+    Stores observation, action, and reward for an agent. Does not have additional
+    fields that are present in AgentExperience.
+    """
+
+    obs: List[np.ndarray]
+    reward: float
+    action: ActionTuple
+    done: bool
+
+
 class AgentExperience(NamedTuple):
```

Comment:
I think making AgentExperience inherit from or have an AgentStatus is a good idea.

Reply:
After some conversation with @chriselion we decided to keep them separate, at least between the DemonstrationProvider and here. 

I'd prefer inheritance but it seems like a rabbit hole that would warrant a deeper think, so my vote is not to do this here (it's already big enough): http://zecong.hu/2019/08/10/inheritance-for-namedtuples/

Ideally I think these should be DataClasses and not named tuples, but until we drop Python 3.6 we won't be able to use them.